src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/__tests__/functions.test.ts

```diff
@@ -208,6 +208,93 @@ describe('functions arg suggestions', () => {
+      ...callbacks,
+      getLicense: jest.fn(async () =>
+        Promise.resolve({
+          hasAtLeast: (license: string) => license.toLowerCase() !== 'platinum',
+        })
+      ),
+    };
+
+    const basicSuggestions = await suggest('FROM index | STATS agg = /', {
+      callbacks: basicLicenseCallbacks,
+    });
+
+    // Should include basic function but not platinum function
+    expect(basicSuggestions.some((s) => s.text.includes('CATEGORIZE'))).toBe(false);
+    expect(basicSuggestions.some((s) => s.text.includes('ST_EXTENT_AGG'))).toBe(true);
+  });
+
+  it('shows all aggregation functions with platinum license', async () => {
+    const { suggest } = await setup();
+    const callbacks = createCustomCallbackMocks();
+
+    const platinumLicenseCallbacks = {
+      ...callbacks,
+      getLicense: jest.fn(async () =>
+        Promise.resolve({
+          hasAtLeast: (license: string) => license.toLowerCase() === 'platinum',
+        })
+      ),
+    };
+
+    const platinumSuggestions = await suggest('FROM index | STATS agg = /', {
+      callbacks: platinumLicenseCallbacks,
+    });
+
+    // Should include all functions
+    expect(platinumSuggestions.some((s) => s.text.includes('CATEGORIZE'))).toBe(true);
+    expect(platinumSuggestions.some((s) => s.text.includes('ST_EXTENT_AGG'))).toBe(true);
+  });
+
+  it('filters scalar functions inside ST_EXTENT_AGG with basic license', async () => {
+    const { suggest } = await setup();
+    const callbacks = createCustomCallbackMocks();
+
+    const basicLicenseCallbacks = {
+      ...callbacks,
+      getLicense: jest.fn(async () =>
+        Promise.resolve({
+          hasAtLeast: (license: string) => license.toLowerCase() !== 'platinum',
+        })
+      ),
+    };
+
+    const partialSuggestions = await suggest('FROM index | STATS agg = ST_EXTENT_AGG( /', {
+      callbacks: basicLicenseCallbacks,
+    });
+
+    // Should include basic function but not platinum function
+    expect(partialSuggestions.some((s) => s.text.includes('TO_CARTESIANPOINT'))).toBe(true);
+    expect(partialSuggestions.some((s) => s.text.includes('TO_CARTESIANSHAPE'))).toBe(false);
+  });
+
+  it('shows all scalar functions inside ST_EXTENT_AGG with platinum license', async () => {
+    const { suggest } = await setup();
+    const callbacks = createCustomCallbackMocks();
+
+    const platinumLicenseCallbacks = {
+      ...callbacks,
+      getLicense: jest.fn(async () =>
+        Promise.resolve({
+          hasAtLeast: (license: string) => license.toLowerCase() === 'platinum',
+        })
+      ),
+    };
+
+    const partialPlatinumSuggestions = await suggest('FROM index | STATS agg = ST_EXTENT_AGG( /', {
+      callbacks: platinumLicenseCallbacks,
+    });
+
+    expect(partialPlatinumSuggestions.some((s) => s.text.includes('TO_CARTESIANPOINT'))).toBe(true);
+    expect(partialPlatinumSuggestions.some((s) => s.text.includes('TO_CARTESIANSHAPE'))).toBe(true);
+  });
```

Comment:
It's great to see these test cases moved here. But I'm still seeing us using real function definitions from Elasticsearch in these tests. Can we replace these with test fixtures using `setTestFunctions`?